From aa40fa0460237e56894920f444c6837d28fda4a7 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 27 Jun 2023 17:04:11 +0300
Subject: [PATCH] [SW-149942] efuse: add init as part of general flow

efuse init was recently removed, as part of a patch
which removes tpm enable:
[SW-142278] h6: tpm: remove tpm_en efuse bit check

efuse init was done in code in the first attempt to read
efuse, hence - since no one reads anything from efuse
(as part of the above patch), there is no init and any other
later project (preboot, mgmt etc) fails in efuse read.

This patch fix this issue by forcing efuse init as part
of general initialization in main.

Change-Id: I2ea459abc045a07fd8c52d5985aae1cc7dd8502e
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index c156478..a44da46 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -19,6 +19,7 @@
 #include "pll.h"
 #include <boot_if.h>
 #include <pci.h>
+#include <efuse.h>
 #include <hl.h>
 
 struct device_ca53_cfg ca53_cfg;
@@ -353,6 +354,7 @@
 		hl_pcie_enable_elbi_access();
 	ppboot_time_critical_section_end();
 	/**********time critical section end************/
+	preboot_fuse_init();
 	ppboot_pcie_drain_init();
 	ppboot_version();
 	rst_src_dump();
